tests/ibc_pfm_test.go

```diff
@@ -0,0 +1,351 @@
+	dymintTomlOverrides["settlement_layer"] = "dymension"
+	dymintTomlOverrides["node_address"] = fmt.Sprintf("http://dymension_100-1-val-0-%s:26657", t.Name())
+	dymintTomlOverrides["rollapp_id"] = "demo-dymension-rollapp"
+
+	configFileOverrides["config/dymint.toml"] = dymintTomlOverrides
+	// Create chain factory with dymension
+	numHubVals := 1
+	numHubFullNodes := 1
+	numRollAppFn := 0
+	numRollAppVals := 1
+	numVals := 1
+	numFullNodes := 0
+	cf := test.NewBuiltinChainFactory(zaptest.NewLogger(t), []*test.ChainSpec{
+		{
+			Name: "rollapp1",
+			ChainConfig: ibc.ChainConfig{
+				Type:                "rollapp-dyms",
+				Name:                "rollapp-test",
+				ChainID:             "demo-dymension-rollapp",
+				Images:              []ibc.DockerImage{rollappImage},
+				Bin:                 "rollappd",
+				Bech32Prefix:        "rol",
+				Denom:               "urax",
+				CoinType:            "118",
+				GasPrices:           "0.0urax",
+				GasAdjustment:       1.1,
+				TrustingPeriod:      "112h",
+				NoHostMount:         false,
+				ModifyGenesis:       nil,
+				ConfigFileOverrides: configFileOverrides,
+			},
+			NumValidators: &numRollAppVals,
+			NumFullNodes:  &numRollAppFn,
+		},
+		{
+			Name:          "dymension-hub",
+			ChainConfig:   dymensionConfig,
+			NumValidators: &numHubVals,
+			NumFullNodes:  &numHubFullNodes,
+		},
+		{
+			Name:          "osmosis",
+			Version:       "v17.0.0",
+			ChainConfig:   osmosisConfig,
+			NumValidators: &numVals,
+			NumFullNodes:  &numFullNodes,
+		},
+	})
+	chains, err := cf.Chains(t.Name())
+	require.NoError(t, err)
+
+	rollapp1 := chains[0].(*dym_rollapp.DymRollApp)
+	dymension := chains[1].(*dym_hub.DymHub)
+	osmosis := chains[2].(*cosmos.CosmosChain)
+
+	_ = rollapp1
+	_ = dymension
+	_ = osmosis
+
+	// Relayer Factory
+	client, network := test.DockerSetup(t)
+	r := test.NewBuiltinRelayerFactory(ibc.CosmosRly, zaptest.NewLogger(t),
+		relayer.CustomDockerImage("ghcr.io/cosmos/relayer", "reece-v2.3.1-ethermint", "100:1000"),
+	).Build(t, client, network)
+
+	ic := test.NewSetup().
+		AddRollUp(dymension, rollapp1).
+		AddChain(osmosis).
+		AddRelayer(r, "relayer").
+		AddLink(test.InterchainLink{
+			Chain1:  dymension,
+			Chain2:  rollapp1,
+			Relayer: r,
+			Path:    pathHubToRollApp,
+		}).
+		AddLink(test.InterchainLink{
+			Chain1:  dymension,
+			Chain2:  osmosis,
+			Relayer: r,
+			Path:    pathDymToOsmos,
+		})
+
+	rep := testreporter.NewNopReporter()
+	eRep := rep.RelayerExecReporter(t)
+
+	err = ic.Build(ctx, eRep, test.InterchainBuildOptions{
+		TestName:         t.Name(),
+		Client:           client,
+		NetworkID:        network,
+		SkipPathCreation: true,
+		// This can be used to write to the block database which will index all block data e.g. txs, msgs, events, etc.
+		// BlockDatabaseFile: test.DefaultBlockDatabaseFilepath(),
+	})
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		_ = ic.Close()
+	})
+
+	// Create relayer rollapp to dym
```

Comment:
Can relayer setup be extracted into some helper function?

Reply:
Also, I don't recall rollapp -> dymension connection had to be done this way in other tests. Why here is it different?

Reply:
In other tests, the `SkipPathCreation`is set to [false](https://github.com/dymensionxyz/e2e-tests/blob/20f0844e4031882bd9d9ca213e015a2c3119ebf3/tests/ibc_timeout_test.go#L102) so that it automatically setup the connections.
But if there is 2 or more connections, it will cause errors so we need set it to true and do create connections mannually